improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -286,7 +295,8 @@ def qsat_differential(qs, t, q, p):
         humidity_in.copy(),
         args=(temperature_in, humidity_in, pressure),
         tol=1e-6,
-        maxiter=6,
+        maxiter=6 if humidity_in.size > 100 else 10,
+        disp=True,
```

Comment:
I cannot find in the documentation a description of how this iteration works with arrays, i.e. whether the temperature_in, humidity_in, and pressure arrays are actually changed in size as the iterations continue. The `humidity_in.size` check only makes sense if the array is actually being resized, rather than the iterator keeping an index of which points in a fixed size array it needs to continue to iterate over.

I've tested this code with and without the change using the global data that Stephen saved and it does indeed fix the issue, suggesting the array must be being resized during the iterations such that `maxiter` is being increased as the array shrinks.

Reply:
Ah. Maybe I should have explained this a little better. The calling function is cycling upwards through the pressure coordinate and at each cycle, only passes in the spatial points that have not already become negatively buoyant (indicating that the top of the convection has been reached).

Whether the iterator ends point A before point B when the converge independently, I don't know.